enhancements/microshift/default-csi-plugin.md

```diff
@@ -0,0 +1,668 @@
+  are deployed to the field, persistent storage will be created automatically, prior to the workload starting.
+* As a device owner, I must be able to protect the host OS, or other workloads, from workloads that may over-consume
+  storage, which could destabilize the platform.
+* As a device owner, I want to configure system and cluster storage parameters pre-boot and have those configurations
+  acted upon when MicroShift starts.
+* As a device owner, I want to update cluster storage configuration with the assurance that a faulty config will not
+  destabilize the system.
+
+### Goals
+
+<!---
+Summarize the specific goals of the proposal. How will we know that
+this has succeeded?  A good goal describes something a user wants from
+their perspective, and does not include the implementation details
+from the proposal.
+--->
+
+* Provide a default, production-ready, dynamic storage solution out of the box.
+* Protect the host OS from greedy workloads to assure system stability.
+* Align with MicroShift's lifecycle management model via rpm-ostree atomic updates.
+
+### Non-Goals
+
+<!---
+What is out of scope for this proposal? Listing non-goals helps to
+focus discussion and make progress. Highlight anything that is being
+deferred to a later phase of implementation that may call for its own
+enhancement.
+--->
+
+* Support multi-node cluster topologies
+* Enable on-disk encryption
+
+## Proposal
+
+<!---
+This is where we get down to the nitty gritty of what the proposal
+actually is. Describe clearly what will be changed, including all of
+the components that need to be modified and how they will be
+different. Include the reason for each choice in the design and
+implementation that is proposed here, and expand on reasons for not
+choosing alternatives in the Alternatives section at the end of the
+document.
+--->
+
+### Workflow Description
+
+<!---
+Explain how the user will use the feature. Be detailed and explicit.
+Describe all of the actors, their roles, and the APIs or interfaces
+involved. Define a starting state and then list the steps that the
+user would need to go through to trigger the feature described in the
+enhancement. Optionally add a
+[mermaid](https://github.com/mermaid-js/mermaid#readme) sequence
+diagram.
+
+Use sub-sections to explain variations, such as for error handling,
+failure recovery, or alternative outcomes.
+
+For example:
+
+**cluster creator** is a human user responsible for deploying a
+cluster.
+
+**application administrator** is a human user responsible for
+deploying an application in a cluster.
+
+1. The cluster creator sits down at their keyboard...
+2. ...
+3. The cluster creator sees that their cluster is ready to receive
+   applications, and gives the application administrator their
+   credentials.
+--->
+
+Device Owner: A human user with privileged access on the RHEL host and cluster-admin access in the MicroShift cluster.
+
```

Comment:
The workflows below are all manual, which helps understand the individual steps. We don't expect users to connect to the console very often, though, so you could include a "disclaimer" here that the steps are very similar for automated flows but involve building the image with the configuration file in it, etc.